hdf5-vol-daos.spec

```diff
@@ -162,6 +164,8 @@ HDF5 VOL DAOS tests with mpich
 %setup -T -D -b 1 -n vol-daos-%{vol_tag}
 %patch0 -p1 -b .34db47e61d48988458a251af7148f15b1ecec8b8.patch
 %patch1 -p1 -b .5eed612db03458c1b21e18f13fa2d1cb51e96193.patch
+%patch2 -p1 -b .e29abaf69233e014e6e26cf927cc55934c178128.patch
```

Comment:
I think applying this patch doesn't work on the vol test like the vol repo since the test is a submodule. so this patching in this PR is wrong.. but im not quite sure how to do it for the vol test. any ideas @phender @mjean308 ?

Reply:
ok i think this works like this to patch the test lib. so once packaging is fixed, i can rebase this PR to test again

Reply:
I would try updating https://github.com/daos-stack/hdf5-vol-daos/tree/master/packaging as part of this PR (run https://github.com/daos-stack/packaging/blob/master/install) in your local repo.  Similar issues where seen in a recent mpifileutils PR where [swapping `yum` commands for `dnf`](https://github.com/daos-stack/packaging/pull/52) fixed issues with resolving dependencies.

Reply:
I think you need to apply the patch while in the test directory
Try this
%prep
%setup -n vol-daos-%{vol_tag}
%setup -T -D -b 1 -n vol-daos-%{vol_tag}
%patch0 -p1 -b .34db47e61d48988458a251af7148f15b1ecec8b8.patch
%patch1 -p1 -b .5eed612db03458c1b21e18f13fa2d1cb51e96193.patch
cd ../vol-tests-%{vol_test_tag}
%patch2 -p1 -b .e29abaf69233e014e6e26cf927cc55934c178128.patch
cd ../vol-daos-%{vol_tag}
mv ../vol-tests-%{vol_test_tag}/* test/vol/

Reply:
yes i had updated to cd back to this dir. please check the latest changes

Reply:
> I would try updating https://github.com/daos-stack/hdf5-vol-daos/tree/master/packaging as part of this PR (run https://github.com/daos-stack/packaging/blob/master/install) in your local repo. Similar issues where seen in a recent mpifileutils PR where [swapping `yum` commands for `dnf`](https://github.com/daos-stack/packaging/pull/52) fixed issues with resolving dependencies.

ok updated.. let's see

Reply:
still failed in resolving dependencies.. at least for  centos7.
for leap15 it failed in finding MPI compiler.
centos 8 passes.

Reply:
It seems like something is missing from this error message:
```
DEBUG util.py:446:  Error: Package: daos-client-1.3.104-1.6816.g5d1e91ee.el7.x86_64 (repo.dc.hpdd.intel.com_repository_daos-stack-centos-7-x86_64-stable-group_)
DEBUG util.py:446:             Requires: daos(x86-64) = 1.3.104-1.6816.g5d1e91ee.el7
```
At the end it ends up stating that its installing the correct dependency (`1.3.104-2.6818`) - despite listing a large number of incorrect ones:
```
...
DEBUG util.py:446:             Available: daos-1.3.104-1.6816.g5d1e91ee.el7.x86_64 (repo.dc.hpdd.intel.com_repository_daos-stack-centos-7-x86_64-stable-group_)
DEBUG util.py:446:                 daos(x86-64) = 1.3.104-1.6816.g5d1e91ee.el7
DEBUG util.py:446:             Available: daos-1.3.104-1.6817.gaf19e7fb.el7.x86_64 (repo.dc.hpdd.intel.com_repository_daos-stack-centos-7-x86_64-stable-group_)
DEBUG util.py:446:                 daos(x86-64) = 1.3.104-1.6817.gaf19e7fb.el7
DEBUG util.py:446:             Installing: daos-1.3.104-2.6818.g2c9c68cc.el7.x86_64 (repo.dc.hpdd.intel.com_repository_daos-stack-centos-7-x86_64-stable-group_)
DEBUG util.py:446:                 daos(x86-64) = 1.3.104-2.6818.g2c9c68cc.el7
```
Maybe the difference is the use of `/usr/bin/dnf builddep` on CentOS 8 vs `/usr/bin/yum-builddep` on CentOS 7

Reply:
well at the begining it says:
daos-client-1.3.104-1.6816 requires daos 1.3.104-1.6816.
but it's installing:
daos 1.3.104-2.6818.g2c9c68cc.el7
but lists 6816 as available.. so not sure if this is correct dependency installed

Reply:
Ah, I missed that!  So we have a newer daos version (104-2.6818) than the detected daos-client version (104-1.6819).  One would think this would be resolved by https://github.com/daos-stack/daos/blob/master/utils/rpms/daos.spec#L177-L179

Reply:
As a data point the CentOS 7 build was successful in https://build.hpdd.intel.com/job/daos-stack/job/hdf5-vol-daos/job/PR-24/6/artifact/artifacts/centos7/root.log/*view*/ with the `1.3.104-3.6829` daos packages.